govc/importx/archive.go

```diff
@@ -127,23 +122,61 @@ func (t *TapeArchive) Open(name string) (io.ReadCloser, int64, error) {
 
 type FileArchive struct {
 	path string
+	Opener
 }
 
 func (t *FileArchive) Open(name string) (io.ReadCloser, int64, error) {
 	fpath := name
 	if name != t.path {
-		fpath = filepath.Join(filepath.Dir(t.path), name)
+		index := strings.LastIndex(t.path, "/")
+		if index != -1 {
+			fpath = t.path[:index] + "/" + name
+		}
 	}
 
-	s, err := os.Stat(fpath)
+	return t.OpenFile(fpath)
+}
+
+type Opener struct {
+}
+
+func isRemotePath(path string) bool {
+	if strings.HasPrefix(path, "http://") || strings.HasPrefix(path, "https://") {
+		return true
+	}
+	return false
+}
+
+func (o Opener) OpenLocal(path string) (io.ReadCloser, int64, error) {
+	f, err := os.Open(path)
 	if err != nil {
 		return nil, 0, err
 	}
 
-	f, err := os.Open(fpath)
+	s, err := f.Stat()
 	if err != nil {
 		return nil, 0, err
 	}
 
 	return f, s.Size(), nil
 }
+
+func (o Opener) OpenFile(path string) (io.ReadCloser, int64, error) {
+	if isRemotePath(path) {
+		return o.OpenRemote(path)
+	}
+	return o.OpenLocal(path)
+}
+
+func (o Opener) OpenRemote(url string) (io.ReadCloser, int64, error) {
+	resp, err := http.Get(url)
```

Comment:
We want to use the command's vim25.Client here, so the TLS config applies (`-k`, `tls-ca-certs`, `-tls-known-hosts` flags, etc.)

Reply:
got it. will update it.